test/test_utils.py

```diff
@@ -112,6 +114,119 @@ def __len__(self):
         return 10
 
 
+class TestCheckpoint(TestCase):
+
+    # Test whether checkpoint is being triggered or not. For this, we check
+    # the number of times forward pass happens
+    def test_checkpoint_trigger(self):
+
+        class Net(nn.Module):
```

Comment:
This class looks weirdly complicated, because it basically has to reimplement `checkpoint_sequential`. A better strategy would be to put the counting in the `Linear`, and then use the canonical implementation of `checkpoint_sequential`.